deploy/lib/deployer/repo.rb

```diff
@@ -0,0 +1,138 @@
+class Deployer
+  class Repo
+    def initialize(name, config:)
+      @name = name
+      @config = config
+    end
+
+    def update_package
+      run
+      log "Successfully updated #{package_name} to #{version} in #{name} (https://github.com/#{owner}/#{name}/pull/#{pr_number})"
+    rescue StandardError => e
+      cleanup
+      raise e
+    end
+
+    attr_reader :name
+
+    private
+
+    attr_reader :config, :pr_number
+
+    def run
+      clone_repo
+      Dir.chdir(name) do
+        create_branch
+        run_upgrade_command
+        commit_and_push_changes
+        create_pr
+        automerge_pr
+      end
+      cleanup
+    end
+
+    def clone_repo
+      url = "github.com/#{owner}/#{name}.git"
+      log "Cloning #{name}"
+      stdout, status =
+        Open3.capture2(
+          "git clone https://#{config.github_token}:x-oauth-basic@#{url} --depth=1"
+        )
+      raise FailedToCloneRepo, stdout unless status.success?
+    end
+
+    def create_branch
+      log "Creating branch #{branch_name}"
+      stdout, status = Open3.capture2("git checkout -b #{branch_name}")
+      raise CreateBranchFailure, stdout unless status.success?
+    end
+
+    def run_upgrade_command
+      log "Running #{upgrade_command}"
+      stdout, status =
+        Open3.capture2("#{upgrade_command} #{package_name}@#{version}")
+
+      raise UpgradeCommandFailure, stdout unless status.success?
+    end
+
+    def commit_and_push_changes
+      stdout, status =
+        Open3.capture2("git commit -am 'bump #{package_name} to #{version}'")
+      raise CommitChangesFailure, stdout unless status.success?
+
+      stdout2, status2 = Open3.capture2("git push origin #{branch_name} -f")
+      raise PushBranchFailure, stdout2 unless status2.success?
+    end
+
+    def create_pr
+      response =
+        client.create_pull_request(
+          "#{owner}/#{name}",
+          "main",
+          branch_name,
+          pr_title,
+          pr_body
+        )
+      raise FailedToCreatePRError, response if response.number.nil?
+
+      @pr_number = response.number
+    end
+
+    def automerge_pr
+      return unless automerge
+
+      log "Merging PR #{pr_number}"
+      stdout, status = Open3.capture2("gh pr merge #{pr_number} --auto --merge")
```

Comment:
I don't think this is an _important_ question, but it seems like `Octokit` and the `gh` command have the same core purpose statement. Is there a particular reason for times when we use one or the other? Just whatever is most convenient?

Reply:
In particular for this command, the `automerge` feature is one where it waits for the checks to succeed and then merges after the checks are complete.  From what I could tell about the octokit command, it just handled complete merging (skipping checks).  For now I wanted to fall back to the the auto merging behavior so that we can follow up with checks.